### PR TITLE
feat(messages): add new message endpoint support

### PR DESCRIPTION
**Pull request by juanazam**
* Added new `GET /messages/:id` endpoint
* Added new `GET /messages/:id/content` endpoint
* Added new `GET /messages/:id/events` endpoint
* Added new `GET /messages/:id/activities` endpoint
* Added new `GET /users/:id/messages` endpoint
* Added new `GET /objects/:collection/:id/messages` endpoint

### CODE REVIEW

File: knockapi/client.py [hunk 1]
```diff
@@ -20,6 +20,7 @@ def request(self, method, endpoint, payload=None):
         r = requests.request(
             method,
             url,
+            params=payload,
```
Comment (juanazam) [inline knockapi/client.py:23]: parameters sent to GET endpoints were not working 😱

Comment (thomaswhyyou) [reply knockapi/client.py:23]: 😬

File: knockapi/resources/users.py [hunk 2]
```diff
@@ -349,3 +349,21 @@ def set_category_preferences(self, user_id, key, setting, options={}):
         params = setting if type(setting) is dict else {'subscribed': setting}
 
         return self.client.request('put', endpoint, payload=params)
+
+    ##
+    # Messages
+    ##
+
+    def get_messages(self, id, options=None):
```
Comment (thomaswhyyou) [inline knockapi/resources/users.py:357]: Is there a difference between passing in an empty dict like we are doing with the `set_category_preferences` func above vs defaulting to `None`? If not, i might just default to an empty dict to keep things consistent.

File: knockapi/resources/objects.py [hunk 3]
```diff
@@ -111,4 +111,23 @@ def set_channel_data(self, collection, id, channel_id, channel_data):
...
         endpoint = '/objects/{}/{}/channel_data/{}'.format(collection, id, channel_id)
-        return self.client.request('put', endpoint, payload={'data': channel_data})
\ No newline at end of file
+        return self.client.request('put', endpoint, payload={'data': channel_data})
+
+    ##
+    # Messages
+    ##
+
+    def get_messages(self, collection, id, options=None):
```
Comment (thomaswhyyou) [inline knockapi/resources/objects.py:120]: Same question as below. Also in `knockapi/resources/messages.py`.

File: knockapi/resources/messages.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+from .service import Service
+
+default_set_id = "default"
```
Comment (thomaswhyyou) [inline knockapi/resources/messages.py:3]: What does this do? There seems to be a few places where we might've copied this over without actually using it.

**Review by thomaswhyyou — COMMENTED** [review]
A couple minor questions/comments

File: knockapi/resources/objects.py (hunk 3 above)
Comment (juanazam) [reply knockapi/resources/objects.py:120]: will update!

File: knockapi/resources/users.py (hunk 2 above)
Comment (juanazam) [reply knockapi/resources/users.py:357]: @thomaswhyyou I copied this from https://github.com/knocklabs/knock-python/blob/464905cad760bf286bfa8c03cb241c8393c123cb/knockapi/resources/users.py#L81 which also supports pagination so I thought it made sense to follow that approach, what do you think?

File: knockapi/resources/objects.py (hunk 3 above)
Comment (juanazam) [reply knockapi/resources/objects.py:120]: @thomaswhyyou actuall, please see my other comment :)

File: knockapi/resources/users.py (hunk 2 above)
Comment (thomaswhyyou) [reply knockapi/resources/users.py:357]: Ah i see. I'm pretty sure they don't make a difference, so given we already default to one or the other, probably fine to leave it :)

File: knockapi/client.py (hunk 1 above)
Comment (cjbell) [reply knockapi/client.py:23]: @juanazam did you test POST payloads here as well? It seems a bit weird now that we're passing both `params=payload` and `json=payload`. I want to make sure this won't break anything.

**Review by cjbell — APPROVED** [review]
One comment on the GET params!

File: knockapi/client.py (hunk 1 above)
Comment (juanazam) [reply knockapi/client.py:23]: @cjbell will check so we are sure!

Comment (juanazam) [reply knockapi/client.py:23]: @cjbell confirmed, tried both PUT and POST and everything goes through.

**Pull request merged**